common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -315,19 +315,15 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](dbProtocol: St
     val start = transid.started(this, LoggingMarkers.DATABASE_QUERY, s"[COUNT] '$dbName' searching '$table")
 
     val f = client
-      .executeView(firstPart, secondPart)(
-        startKey = startKey,
-        endKey = endKey,
-        skip = Some(skip),
-        stale = stale,
-        reduce = true)
+      .executeView(firstPart, secondPart)(startKey = startKey, endKey = endKey, stale = stale, reduce = true)
       .map {
         case Right(response) =>
           val rows = response.fields("rows").convertTo[List[JsObject]]
 
-          val out = if (!rows.isEmpty) {
+          val out = if (rows.nonEmpty) {
             assert(rows.length == 1, s"result of reduced view contains more than one value: '$rows'")
-            rows.head.fields("value").convertTo[Long]
+            val count = rows.head.fields("value").convertTo[Long]
+            if (count > skip) count - skip else 0L
```

Comment:
skip may be negative also - although it is better to reject.a negative skip at the API handler (validate the value there) but defensively code against that as well here?

Reply:
Makes sense. Added a check for both `limit` and `skip` to  be non negative in both `query` and `count`

Reply:
@rabbah, are you saying to terminate a route if those values are negative at the top API level?

Reply:
yes - we can just validate and reject early (we do some other validations, so could extend to count and skip)... orthogonal to this PR.

Reply:
Make sense. With these changes, `There was an internal server error.` is returned when a negative `skip` value is specified. A content-type of `text/plain; charset=UTF-8` is returned as well.